hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/DefaultWriteBehindProcessor.java

```diff
@@ -243,14 +244,22 @@ private Map convertToObject(Map<Object, DelayedEntry> batchMap) {
             public boolean run() throws Exception {
                 callBeforeStoreListeners(batchMap.values());
                 final Map map = convertToObject(batchMap);
-                final boolean result = operationType.processBatch(map, mapStore);
+                boolean result;
+                try {
+                    result = operationType.processBatch(map, mapStore);
+                } catch (Exception ex) {
+                    Iterator<Object> keys = batchMap.keySet().iterator();
+                    while (keys.hasNext()) {
```

Comment:
I agree. It's a behavior change and we should avoid this in a maintenance release unless the old behavior is a clear bug. 
the [forward port is OK to me](https://github.com/hazelcast/hazelcast/pull/9747/files), assuming it also described the behavior in the contract of MapStore. 

@dsukhoroslov: the 3.8 is going to be out soon.